src/main/java/jewellery/inventory/service/ProductService.java

```diff
@@ -77,6 +78,10 @@ public ProductReturnResponseDto getProductReturnResponseDto(
     return productMapper.mapToProductReturnResponseDto(sale, product);
   }
 
+  public BigDecimal getProductSalePrice(Product product){
+    return productMapper.mapToProductResponseDto(product).getSalePrice();
```

Comment:
Let's make `ProductMapper::calculateTotalPrice(Product product)` public and use it instead of mapping the whole product just to get the price.